packages/nextjs/components/NetworksDropdown.tsx

```diff
@@ -44,14 +60,34 @@ const groupedOptions = networks.reduce<GroupedOptions>(
     mainnet: { label: "mainnet", options: [] },
     testnet: { label: "testnet", options: [] },
     localhost: { label: "localhost", options: [] },
+    other: { label: "other", options: [] },
   },
 );
 
+const allChains = Object.values(chains)
+  .filter(
+    chain =>
+      !networks.some(network => network.id === chain.id) &&
+      !Object.values(groupedOptions).some(group => group.options.some(option => option.value === chain.id)),
```

Comment:
Instead of .some, can't we have just did : 

`!networks.include(chain.id)` ? 

Also for second condition, not sure if its best idea to extra the inner function in for better redability. some and name them so that its easier to read ? lol also noticed it's a nested loop but I think it's fine since groupedOptions is small